design/go-api.md

```diff
@@ -0,0 +1,72 @@
+# Replace the current typescript kurl.sh api for creating installers with a go api using gin
+ 
+This proposal outlines the benefits and challenges of replacing the current typescript api with a more simple golang api.
+
+## Goals
+
+- Significantly reduce code complexety around making changes to the installer type
+- Provide type safety and validation for POSTed installers
+
+## Non Goals
+
+- Create an API for kurlkinds dsecriptions. This probably best suited in the kurl.sh site repo as it is not core functionality of the api.
+
+## Background
+
+Previously, kURL has used the backend typescript api to take a yaml config and make an install script with options and bash variables set for the install.
+With the recent changes to kURL such as the addition of the CRD, the current install script uses go binaries to parse and traverse the yaml tree to set options.
+
+We can greatly reduce code complexity from the no longer used parts of the typescript api and gain added type safety by moving this api to Go.
+
+## High-Level Design
+
+Endpoints
```

Comment:
The kurl.sh frontend needs the `GET /installer` endpoint to list available versions

Reply:
what is this used for currently?

Reply:
For the dropdowns on the homepage of kurl.sh

Reply:
I think we might want to move this to kurl.sh instead